clients/geth/specular/rollup/services/sequencer/sequencer.go

```diff
@@ -159,14 +179,20 @@ func (s *Sequencer) sequencingLoop(genesisRoot common.Hash) {
 				continue
 			}
 			// Update queued assertion to latest batch
-			queuedAssertion.VmHash = batch.LastBlockRoot()
-			queuedAssertion.CumulativeGasUsed.Add(queuedAssertion.CumulativeGasUsed, batch.GasUsed)
-			queuedAssertion.InboxSize.Add(queuedAssertion.InboxSize, batch.InboxSize())
-			queuedAssertion.EndBlock = batch.LastBlockNumber()
+			queuedAssertion.VmHash = combinedBatch.LastBlockRoot()
+			queuedAssertion.CumulativeGasUsed.Add(queuedAssertion.CumulativeGasUsed, combinedBatch.GasUsed)
+			queuedAssertion.InboxSize.Add(queuedAssertion.InboxSize, combinedBatch.InboxSize())
+			queuedAssertion.EndBlock = combinedBatch.LastBlockNumber()
 			// If no assertion is pending, commit it
 			if pendingAssertion == nil {
 				commitAssertion()
 			}
+			batchTxs = nil
+			combinedBatch = nil
```

Comment:
probably makes more sense to just declare `var combinedBatch *rollupTypes.TxBatch` in the case (then it'll also be GC'd automatically w/o assigning nil)